app/models/work_group.rb

```diff
@@ -64,18 +73,18 @@ def minified_url(user)
     user.update(password: one_time_password)
     url_suffix = "/classrooms/#{classroom_id}/work_groups/#{id}?email=#{user.email}&password=#{one_time_password}"
     if Rails.env == 'production'
-      url = 'https://odaiba-app.netlify.app' + url_suffix
+      url = "https://odaiba-app.netlify.app#{url_suffix}"
       LinkShortener.new(url).call
     else
-      'http://localhost:3000' + url_suffix
+      "http://localhost:3000#{url_suffix}"
     end
   end
 
   def assign_colors
     # orange, yellow, green, blue, pink, purple
-    colors = ['#FF5733', '#FFE632', '#11841F', '#2B86E7', '#E72BD8', '#B554FF']
-    users.each_with_index do |student, index|
-      student.update(pen_colour: colors[index])
+    sample_colors = COLOURS.to_a.sample(6)
+    student_work_groups.each_with_index do |student, index|
```

Comment:
technically you aren't updating a student but astudent work group
would be better to rename it for clarity sake, though you can use abbreviations